lib/ansible/module_utils/facts/system/distribution.py

```diff
@@ -440,6 +441,18 @@ def parse_distribution_file_ClearLinux(self, name, data, path, collected_facts):
             clear_facts['distribution_release'] = release.groups()[0]
         return True, clear_facts
 
+    def parse_distribution_file_CentOS(self, name, data, path, collected_facts):
+        centos_facts = {}
+        if 'CentOS Linux' in data:
+            centos_facts['distribution'] = 'CentOS'
+        elif 'CentOS Stream' in data:
+            centos_facts['distribution'] = 'CentOS Stream'
```

Comment:
has anyone checked what RHEL 8 returns?

Reply:
@konstruktoid 

> In my opinion centos_facts['distribution'] is correct since CentOS Stream is the new name for the current CentOS distribution.

I don't disagree this is _technically_ correct. But changing this to be _technically_ accurate will break a lot of things and not add much benefit to users in practice.

For instance, the most common use of `ansible_facts.distrition` is for contitionals and including files. If we changed the value from "CentOS" to "CentOS Stream", every conditional using `when: ansible_facts.distribution == 'CentOS'` would have to change. Including files based on the distribution name would also have to be updated. This would also make it a bit more difficult to target "anything that is CentOS".

If the main goal is to be able to detect CentOS Stream vs Core, leaving `ansible_facts.distribution` as "CentOS" and adding an additional field indicating it is Stream is ultimately more useful.

Reply:
RHEL 8:
```
    "ansible_facts": {
        "ansible_distribution": "RedHat",
        "ansible_distribution_file_parsed": true,
        "ansible_distribution_file_path": "/etc/redhat-release",
        "ansible_distribution_file_search_string": "Red Hat",
        "ansible_distribution_file_variety": "RedHat",
        "ansible_distribution_major_version": "8",
        "ansible_distribution_release": "Ootpa",
        "ansible_distribution_version": "8.2"
    },
```

```
> cat /etc/os-release
NAME="Red Hat Enterprise Linux"
VERSION="8.2 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.2"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.2 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8.2:GA"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.2
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.2"

> cat /etc/redhat-release
Red Hat Enterprise Linux release 8.2 (Ootpa)
```

Reply:
Okay, I will rework this to update `distribution_release` for Stream instead!

Reply:
Thanks Rick!

Reply:
@relrod should we do the same for RHEL 8? seems to use the 'version name' right now

Reply:
@bcoca I believe that `distribution_release` currently shows the right thing for RHEL 8, what would it change to?

Reply:
just trying to be congruent, since RHEL8 is also a 'stream release' should this be the release name or the release type? Is 'stream' not just the type but also the new release name for CentOS8? Do we need a new fact to indicate release type?

Reply:
CentOS Stream is just the name of the distro, yeah. It's a "stream" in the sense that it's rolling release. I'm not sure if that's useful to expose in facts or not, if it is I'd consider it a separate feature.

But I think we're currently doing the right thing for RHEL (displaying the name) and for CentOS (as of this PR).